### PR TITLE
glsl-out: fixes

### PR DESCRIPTION
**Pull request by JCapucho**
Fixes `UnaryOp::Not` generating the wrong op (`~` instead of `!`) on booleans and correctly handles storage images

### CODE REVIEW

File: src/back/glsl.rs [hunk 1]
```diff
@@ -251,9 +251,43 @@ pub fn write<'a>(module: &'a Module, out: &mut impl Write, options: Options) ->
             })?
         };
 
-        if let Some(ref binding) = global.binding {
-            if !es {
-                write!(out, "layout({}) ", Binding(binding))?;
+        let format = match module.types[global.ty].inner {
```
Comment (kvark) [inline src/back/glsl.rs:254]: nit: call it `storage_format`

File: src/back/glsl.rs [hunk 2]
```diff
@@ -251,9 +251,43 @@ pub fn write<'a>(module: &'a Module, out: &mut impl Write, options: Options) ->
...
+            TypeInner::Image {
+                class: ImageClass::Storage(format, _),
+                ..
+            } => Some(format),
+            _ => None,
+        };
+
+        if format.is_some() || global.binding.is_some() {
```
Comment (kvark) [inline src/back/glsl.rs:262]: if it's a storage texture type, it has to have a binding. I think we can omit the `format.is_some()` check

File: src/back/glsl.rs [hunk 3]
```diff
@@ -499,7 +533,7 @@ fn write_statement<'a, 'b>(
                     )?;
                 }
 
-                if fallthrough.is_some() {
+                if fallthrough.is_none() {
```
Comment (kvark) [inline src/back/glsl.rs:536]: heh

File: src/back/glsl.rs [hunk 4]
```diff
@@ -874,25 +911,47 @@ fn write_expression<'a, 'b>(
...
-            } else {
-                todo!()
+            let expr = match class {
+                ImageClass::Sampled | ImageClass::Multisampled => {
+                    let ms = match class {
+                        crate::ImageClass::Multisampled => true,
+                        _ => false,
+                    };
+
+                    //TODO: fix this
```
Comment (kvark) [inline src/back/glsl.rs:921]: is this `TODO` still needed?

File: src/back/glsl.rs [hunk 5]
```diff
@@ -874,25 +911,47 @@ fn write_expression<'a, 'b>(
...
+                    let sampler_constructor = format!(
+                        "{}sampler{}{}{}({})",
+                        match kind {
+                            ScalarKind::Sint => "i",
```
Comment (kvark) [inline src/back/glsl.rs:925]: this little piece is repeated at least twice, asks to be moved out :)

File: src/back/glsl.rs (hunk 4 above)
Comment (JCapucho) [reply src/back/glsl.rs:921]: This part will be removed when we implement combined image samplers

File: src/back/glsl.rs (hunk 5 above)
Comment (JCapucho) [reply src/back/glsl.rs:925]: This part will be removed when we implement combined image samplers

File: src/back/glsl.rs [hunk 6]
```diff
@@ -251,9 +251,43 @@ pub fn write<'a>(module: &'a Module, out: &mut impl Write, options: Options) ->
...
+        if global.binding.is_some() {
+            write!(out, "layout(")?;
+
+            if let Some(ref binding) = global.binding {
+                if !es {
+                    write!(out, "{}", Binding(binding))?;
+                }
+            }
+
+            if storage_format.is_some() && global.binding.is_some() {
```
Comment (LingMan) [inline src/back/glsl.rs:271]: This is already inside an `if global.binding.is_some()` block. No need to check twice.

**Pull request merged**